lib/ASTPrinter/ASTStmtPrinter.cpp

```diff
@@ -0,0 +1,111 @@
+#include "ASTPrinter.hpp"
+
+namespace glu::ast {
+
+void ASTPrinter::beforeVisit(ASTNode *node)
+{
+    out.indent(_indent);
+    out << node->getKind()
+        << " at file : " << _srcManager->getBufferName(node->getLocation())
+        << " line : " << _srcManager->getSpellingLineNumber(node->getLocation())
+        << " col : "
+        << _srcManager->getSpellingColumnNumber(node->getLocation());
+    _indent += 2;
+}
+
+void ASTPrinter::afterVisit()
+{
+    _indent -= 2;
+}
+
+/// @brief Visits an AST node.
+/// @param node The AST node to be visited.
+void ASTPrinter::visitASTNode(ASTNode *node)
+{
+    if (!node) {
+        out.indent(_indent);
+        out << "Null ASTNode\n";
+    } else {
+        beforeVisit(node);
+        out << "\n";
+        afterVisit();
+    }
+}
+
+/// @brief Visits an IfStmt node.
+/// @param node The IfStmt node to be visited.
+void ASTPrinter::visitIfStmt(IfStmt *node)
+{
+    beforeVisit(node);
+    out << "\n";
+    visit(node->getCondition());
+    visit(node->getBody());
+    visit(node->getElse());
```

Comment:
visiting the children should be factored in an ASTVisitor subtype: ASTWalker
That's something we'll want to do often, we don't want to write this code each time.